image/image_linux.go

```diff
@@ -400,6 +400,15 @@ var setupSeed = func(tsto *tooling.ToolingStore, model *asserts.Model, opts *Opt
 			return err
 		}
 
+		// Its a bit more tricky to deal with local snaps, as we only have that specific revision
+		// available. Therefore the revision in the local snap must be exactly the revision specified
+		// in the manifest. If it's not, we fail.
+		specifiedRevision := opts.Revisions[info.SnapName()]
```

Comment:
what happens if the file is fully local?  do we save x1 in the manifest? in that case this code is not correct because info.Revision is set to x1 only below by InfoDerived. I think if specifiedRevision is x1 (snap.R(-1)) we should accept info.Revision to be unset at this point

Reply:
Ah right, nice catch, you were absolutely correct, the check itself was completely wrong, and making this check after InfoDervied made it work as expected.